nixos/modules/services/networking/sozu.nix

```diff
@@ -0,0 +1,299 @@
+            default = false;
+            example = true;
+            description = ''
+              Indicates if worker process will be pinned on a core. If you
+              activate this, be sure that you do not have more workers than CPU
+              cores (and leave at least one core for the kernel and the master
+              process).
+            '';
+          };
+
+          max_connections = mkOption {
+            type = types.int;
+            default = 10000;
+            example = 500;
+            description = ''
+              Maximum number of connections to a worker. If it reached that
+              number and there are new connections available, the worker will
+              accept and close them immediately to indicate it is too busy to
+              handle traffic.
+              Defaults to 10000 maximum connections.
+            '';
+          };
+
+          max_buffers = mkOption {
+            type = types.int;
+            default = 1000;
+            example = 500;
+            description = ''
+              Maximum number of buffers used by the protocol implementations for
+              active connections (ie currently serving a request).
+              For now, you should estimate that
+              <literal>max_buffers = number of concurrent requests * 2</literal>.
+              Defaults to 1000.
+            '';
+          };
+
+          buffer_size = mkOption {
+            type = types.int;
+            default = 16384;
+            description = ''
+              Size of the buffers used by the protocol implementations. Each worker will
+              preallocate max_buffers * 2 * buffer_size bytes, so you should plan for this
+              memory usage. If you plan to use sozu's runtime upgrade feature, you should
+              leave enough memory for one more worker (also for the kernel, etc), so total
+              RAM should be larger than <literal>(worker count + 1) * max_buffers * 2 * buffer_size bytes</literal>.
+              Defaults to 16384.
+            '';
+          };
+
+          ctl_command_timeout = mkOption {
+            type = types.int;
+            default = 1000;
+            description = ''
+              How much time (in milliseconds) sozuctl will wait for a command to complete.
+              Defaults to 1000 milliseconds.
+            '';
+          };
+
+          pid_file_path = mkOption {
+            default = "/run/sozu/sozu.pid";
+            type = types.str;
+            description = ''
+              PID file is a file containing the PID of the master process of sozu.
+              It can be helpful to help systemd or any other service system to keep track
+              of the main process across upgrades. PID file is not created unless this option
+              is set or if SOZU_PID_FILE_PATH environment variable was defined at build time.
+            '';
+          };
+
+          tls_provider = mkOption {
+            type = types.str;
```

Comment:
Please use an enum with the two types.

Reply:
how do you mean?

Reply:
something like 
https://github.com/nix-community/home-manager/blob/db00b39a9abec04245486a01b236b8d9734c9ad0/modules/accounts/email.nix#L191